pennylane/transforms/qcut.py

```diff
@@ -293,7 +294,7 @@ def fragment_graph(graph: MultiDiGraph) -> Tuple[Tuple[MultiDiGraph], MultiDiGra
             graph_copy.remove_edge(node1, node2, key=wire)
 
     subgraph_nodes = weakly_connected_components(graph_copy)
-    subgraphs = tuple(graph_copy.subgraph(n) for n in subgraph_nodes)
+    subgraphs = tuple(MultiDiGraph(graph_copy.subgraph(n)) for n in subgraph_nodes)
```

Comment:
The copying is needed because `graph_copy.subgraph(n)` is a frozen graph, but we later on need to remove nodes from these subgraphs.